geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -94,48 +95,37 @@ object GeoMesaSpark {
-            override def load(key: String): SimpleFeatureType = {
-              val spec = System.getProperty(GeoMesaSpark.typeProp(key))
-              if (spec == null) throw new IllegalArgumentException(s"Couldn't find property geomesa.types.$key")
-              SimpleFeatureTypes.createType(key, spec)
-            }
-          })
-
-        val encoderCache = CacheBuilder.newBuilder().build(
-          new CacheLoader[String, AvroFeatureEncoder] {
-            override def load(key: String): AvroFeatureEncoder = new AvroFeatureEncoder(typeCache.get(key))
-          })
-
-        val decoderCache = CacheBuilder.newBuilder().build(
-          new CacheLoader[String, AvroFeatureDecoder] {
-            override def load(key: String): AvroFeatureDecoder = new AvroFeatureDecoder(typeCache.get(key))
-          })
-
-        override def write(kryo: Kryo, out: Output, feature: AvroSimpleFeature): Unit = {
-          val typeName = feature.getFeatureType.getTypeName
-          val len = typeName.length
-          out.writeInt(len, true)
-          out.write(typeName.getBytes(StandardCharsets.UTF_8))
-          val bytes = encoderCache.get(typeName).encode(feature)
-          out.writeInt(bytes.length, true)
-          out.write(bytes)
-        }
-
-        override def read(kry: Kryo, in: Input, clazz: Class[AvroSimpleFeature]): AvroSimpleFeature = {
-          val len = in.readInt(true)
-          val typeName = new String(in.readBytes(len), StandardCharsets.UTF_8)
-          val sft = typeCache.get(typeName)
-          val flen = in.readInt(true)
-          val bytes = in.readBytes(flen)
-          decoderCache.get(typeName).decode(bytes)
-        }
-      })
+    val serializer = new com.esotericsoftware.kryo.Serializer[SimpleFeature]() {
+      val typeCache = CacheBuilder.newBuilder().build(
+        new CacheLoader[String, SimpleFeatureType] {
+          override def load(key: String): SimpleFeatureType = {
+            val spec = System.getProperty(GeoMesaSpark.typeProp(key))
+            if (spec == null) throw new IllegalArgumentException(s"Couldn't find property geomesa.types.$key")
+            SimpleFeatureTypes.createType(key, spec)
+          }
+        })
+
+      val serializerCache = CacheBuilder.newBuilder().build(
+        new CacheLoader[String, SimpleFeatureSerializer] {
+          override def load(key: String): SimpleFeatureSerializer = new SimpleFeatureSerializer(typeCache.get(key))
+        })
+
+
+      override def write(kryo: Kryo, out: Output, feature: SimpleFeature): Unit = {
+        val typeName = feature.getFeatureType.getTypeName
+        out.writeString(typeName)
+        serializerCache.get(typeName).write(kryo, out, feature)
+      }
+
+      override def read(kry: Kryo, in: Input, clazz: Class[SimpleFeature]): SimpleFeature = {
+        val typeName = in.readString()
+        serializerCache.get(typeName).read(kryo, in, clazz)
+      }
+    }
+
+    KryoFeatureSerializer.setupKryo(kryo, serializer)
```

Comment:
I believe so, the KryoFeatureSerializer seems to be setup to work with Avro Features (it registers itself to serialize avro features)